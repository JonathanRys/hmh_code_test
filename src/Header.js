import React, { Component } from 'react'

import PropTypes from 'prop-types'

class Header extends Component {
    render() {
        return (
            <div className="header">
                <h1>{ this.props.firstName? `Hello ${this.props.firstName}`: "Welcome"}</h1>
            </div>
        )
    }
}

Header.propTypes = {
    firstName: PropTypes.string
}

export default Header
