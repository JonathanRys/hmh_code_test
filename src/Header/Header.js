import React, { Component } from 'react'
import './Header.css'
import PropTypes from 'prop-types'

class Header extends Component {
    render() {
        return (
            <div className="header">
                <h1>{this.props.name ? `Hello ${this.props.name}` : "Welcome"}</h1>
            </div>
        )
    }
}

Header.propTypes = {
    name: PropTypes.string
}

export default Header
