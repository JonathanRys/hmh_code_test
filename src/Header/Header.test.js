import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header'

describe('Test Header component', () => {
    it('should render', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header />, div);
    });

    it('should render with name', () => {
        const div = document.createElement('div');
        expect(ReactDOM.render(<Header name="Lester" />, div)).toMatchSnapshot()
    });
});