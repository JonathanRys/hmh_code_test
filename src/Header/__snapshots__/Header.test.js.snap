// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component should render with name 1`] = `
Header {
  "_reactInternalFiber": FiberNode {
    "_debugID": 8,
    "_debugIsCurrentlyTiming": false,
    "_debugOwner": null,
    "_debugSource": null,
    "alternate": null,
    "child": FiberNode {
      "_debugID": 9,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": [Circular],
      "_debugSource": null,
      "alternate": null,
      "child": FiberNode {
        "_debugID": 10,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": [Circular],
        "_debugSource": null,
        "alternate": null,
        "child": null,
        "effectTag": 0,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "internalContextTag": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "children": "Hello Lester",
        },
        "memoizedState": null,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": [Circular],
        "sibling": null,
        "stateNode": <h1>
          Hello Lester
        </h1>,
        "tag": 5,
        "type": "h1",
        "updateQueue": null,
      },
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "internalContextTag": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": <h1>
          Hello Lester
        </h1>,
        "className": "header",
      },
      "memoizedState": null,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": [Circular],
      "sibling": null,
      "stateNode": <div
        class="header"
      >
        <h1>
          Hello Lester
        </h1>
      </div>,
      "tag": 5,
      "type": "div",
      "updateQueue": null,
    },
    "effectTag": 1,
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "internalContextTag": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {
      "name": "Lester",
    },
    "memoizedState": null,
    "nextEffect": null,
    "pendingProps": null,
    "ref": null,
    "return": FiberNode {
      "_debugID": 6,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "alternate": FiberNode {
        "_debugID": 6,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "alternate": [Circular],
        "child": null,
        "effectTag": 0,
        "expirationTime": 1,
        "firstEffect": null,
        "index": 0,
        "internalContextTag": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <div
              class="header"
            >
              <h1>
                Hello Lester
              </h1>
            </div>
          </div>,
          "context": Object {},
          "current": [Circular],
          "finishedWork": null,
          "hydrate": false,
          "isReadyForCommit": false,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingContext": null,
          "remainingExpirationTime": 0,
        },
        "tag": 3,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "callbackList": null,
          "expirationTime": 1,
          "first": Object {
            "callback": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "nextCallback": null,
            "partialState": Object {
              "element": <Header
                name="Lester"
              />,
            },
          },
          "hasForceUpdate": false,
          "isInitialized": false,
          "isProcessing": false,
          "last": Object {
            "callback": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "nextCallback": null,
            "partialState": Object {
              "element": <Header
                name="Lester"
              />,
            },
          },
        },
      },
      "child": [Circular],
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "internalContextTag": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <Header
          name="Lester"
        />,
      },
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": <div>
          <div
            class="header"
          >
            <h1>
              Hello Lester
            </h1>
          </div>
        </div>,
        "context": Object {},
        "current": [Circular],
        "finishedWork": null,
        "hydrate": false,
        "isReadyForCommit": false,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingContext": null,
        "remainingExpirationTime": 0,
      },
      "tag": 3,
      "type": null,
      "updateQueue": null,
    },
    "sibling": null,
    "stateNode": [Circular],
    "tag": 2,
    "type": [Function],
    "updateQueue": null,
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "props": Object {
    "name": "Lester",
  },
  "refs": Object {},
  "state": null,
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
